contracts/tests/psp1155.rs

```diff
@@ -48,6 +48,8 @@ mod psp1155 {
     pub struct PSP1155Struct {
         #[PSP1155StorageField]
         psp1155: PSP1155Data,
+        // fields for hater logic
+        hated_account: AccountId,
```

Comment:
Better to use simple logic in tests. For example, you can add a bool `return_err_on_before`. And return an error)

Also, add test for `after_token_transfer`. Apply the same changes to all tests.